src/global-firestore-data/override-fulfillment-choices.ts

```diff
@@ -1,20 +1,41 @@
 import { overriddenFulfillmentChoicesCollection } from '../firebase-frontend-config';
 import store from '../store';
 
+export const updateRequirementChoices = (
+  updater: (
+    oldChoices: FirestoreOverriddenFulfillmentChoices
+  ) => FirestoreOverriddenFulfillmentChoices
+): void => {
+  overriddenFulfillmentChoicesCollection
+    .doc(store.state.currentFirebaseUser.email)
+    .set(updater(store.state.overriddenFulfillmentChoices));
+};
+
 export const updateRequirementChoice = (
   courseUniqueID: string | number,
-  choiceUpdater: (choice: FirestoreCourseOptInOptOutChoices) => FirestoreCourseOptInOptOutChoices
-): void => {
-  overriddenFulfillmentChoicesCollection.doc(store.state.currentFirebaseUser.email).set({
-    ...store.state.overriddenFulfillmentChoices,
-    [courseUniqueID]: choiceUpdater(
+  updater: (choice: FirestoreCourseOptInOptOutChoices) => FirestoreCourseOptInOptOutChoices
+): void =>
+  updateRequirementChoices(choices => ({
+    ...choices,
+    [courseUniqueID]: updater(
       store.state.overriddenFulfillmentChoices[courseUniqueID] || {
         arbitraryOptIn: {},
         acknowledgedCheckerWarningOptIn: [],
         optOut: [],
       }
     ),
-  });
+  }));
```

Comment:
Refactored this function to use `updateRequirementChoices`, modeling the code in other files (like `global-firestore-data/semesters.ts`)